src/glmtools.jl

```diff
@@ -20,57 +180,157 @@ end
 function mueta{T<:Real}(::CloglogLink, η::T)
     max(eps(T), exp(η) * exp(-exp(η)))
 end
+function inverselink(::CloglogLink, η)
+    expη = exp(η)
+    μ = -expm1(-expη)
+    omμ = exp(-expη)   # the complement, 1 - μ
+    μ, max(realmin(μ), expη * omμ), max(realmin(μ), μ * omμ)
+end
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = η
-mueta(::IdentityLink, η) = 1
+mueta(::IdentityLink, η) = one(η)
+inverselink(::IdentityLink, η) = η, one(η), oftype(η, NaN)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = inv(η)
 mueta(::InverseLink, η) = -inv(abs2(η))
+function inverselink(::InverseLink, η)
+    μ = inv(η)
+    μ, -abs2(μ), oftype(μ, NaN)
+end
+
+linkfun(::InverseSquareLink, μ) = inv(abs2(μ))
+linkinv(::InverseSquareLink, η) = inv(sqrt(η))
```

Comment:
These changes have a small but reproducible impact on the performance for important cases like logistic regression
```sh
bates@thin206:~/.julia/v0.6/GLM⟫ julia perf/glm.jl  # prior to change
  5.074595 seconds (158 allocations: 1.136 GiB, 2.63% gc time)
bates@thin206:~/.julia/v0.6/GLM⟫ julia perf/glm.jl
  5.036847 seconds (158 allocations: 1.136 GiB, 2.65% gc time)
bates@thin206:~/.julia/v0.6/GLM⟫ julia perf/glm.jl
  5.038499 seconds (158 allocations: 1.136 GiB, 2.68% gc time)
```